### DISCUSSION TITLE
can you explain the commercial license

### DISCUSSION THREAD

**Discussion by spxneo** [windmill-labs/windmill#3408] [Questions] [answered]
I thought this was a free open source project that I could host without restrictions but seems like we have to pay for a commercial license and it seems to be not fixed but dependent on # of servers its deployed on???

is there anything like windmill out there without a restrictive license

**Comment by rubenfiszel** [comment, marked as answer]
It is free and open-source without restrictions on the number of servers/workers. However, some features which you would care only if you are an enterprise are enterprise only.

Windmill is the most advanced solutions out there for distributed task queue and workflows because we can be a full team of software engineers working on it since some customers pay for the enterprise edition.

**Reply by mirql** [reply]
![image](https://github.com/user-attachments/assets/b07c8b35-13cf-44b1-a532-42b5949c5236)

Hello, so, you think that "Number of workspaces", "groups", "users", "private hubs", "logs" is enterprise only needs? thats strange, or "global css" :thinking: :man_shrugging: 

Registries/otel/alerts everything is disabled without "EE key", but that is not "EE" stuff its just self-hosted functionality that i dont need to use some third-party resources/computation/saas etc, heart breaking, so sad that good looking and feeling apps are just mimicin to be opensource but defacto is not

![image](https://github.com/user-attachments/assets/900a1a1a-1405-45e3-893b-8d2ff24b5c9a)


please update your repo/docs to not waste time of users with this bait to buy "EE" key for capabilities that dont need any third-party services 

:cry: :broken_heart:

**Reply by mirql** [reply]
wow, also "our pricing explain" in faq also not mention motivation for pricings for self-hosted variant :open_mouth:

**Reply by mirql** [reply]
put this critical info on your github readme main page please, because now it feels like you have some kind of open-source nice app that everybody can self-host and be happy, but its not, just cheap baits, this image please:

![image](https://github.com/user-attachments/assets/626d4ed1-ccd0-4f46-b166-817c9cf74b82)

**Comment by spxneo** [comment]
okay but there is a sliding form in the self-hosted pricing section
